src/cli/javascript/worker.rs

```diff
@@ -0,0 +1,190 @@
+use std::sync::Arc;
+
+use hyper::body::Bytes;
+use mini_v8::Values;
+
+use super::serde_v8::SerdeV8;
+use super::sync_v8::{SyncV8, SyncV8Function};
+use crate::channel::{JsRequest, JsResponse};
+use crate::http::Response;
+use crate::{blueprint, HttpIO, ToAnyHow};
+
+pub struct Worker {
+    sync_v8: SyncV8,
+    on_event_js: SyncV8Function,
+}
+
+fn create_closure(script: &str) -> String {
+    format!("(function() {{{} return onEvent}})();", script)
+}
+
+impl Worker {
+    pub async fn new(script: blueprint::Script, http: impl HttpIO) -> anyhow::Result<Self> {
+        let sync_v8 = SyncV8::new();
+        super::shim::init(&sync_v8, Arc::new(http)).await?;
+        let v8 = sync_v8.clone();
+        let closure = sync_v8
+            .clone()
+            .borrow_ret(move |mv8| {
+                let script = mini_v8::Script {
+                    source: create_closure(script.source.as_str()),
+                    timeout: script.timeout,
+                    ..Default::default()
+                };
+                let value: mini_v8::Value = mv8
+                    .eval(script)
+                    .map_err(|e| anyhow::anyhow!(e.to_string()))?;
+                let closure = value
+                    .as_function()
+                    .ok_or_else(|| anyhow::anyhow!("expected an 'onEvent' function"))?
+                    .clone();
+
+                Ok(v8.as_sync_function(closure))
+            })
+            .await?;
+        Ok(Self { sync_v8: sync_v8.clone(), on_event_js: closure })
+    }
+
+    pub async fn on_event(&self, request: reqwest::Request) -> anyhow::Result<Response<Bytes>> {
+        let (tx, mut rx) = tokio::sync::broadcast::channel::<serde_json::Value>(1024);
```

Comment:
why using broadcast channel? it's many to many channel. Probably, oneshot is enough for single

Reply:
We need to await asynchronously, i don't think its possible with one shot. Will check again.